Number_types/include/CGAL/Quotient.h

```diff
@@ -989,48 +989,54 @@ template < class NT > class Real_embeddable_traits_quotient_base< Quotient<NT> >
           return std::make_pair(l, u);
         }
 
+        std::pair<double, double> interval_from_cpp_rational( const Type& x ) const {
+
+          // Seems fast enough because this conversion happens
+          // only a few times during the run, at least for NEF.
+          boost::multiprecision::cpp_rational rat;
+          CGAL_assertion(!CGAL::is_zero(x.den));
+          if (CGAL::is_negative(x.den)) {
+            rat = boost::multiprecision::cpp_rational(-x.num, -x.den);
+          } else {
+            CGAL_assertion(CGAL::is_positive(x.den));
+            rat = boost::multiprecision::cpp_rational( x.num,  x.den);
+          }
+
+          double l, u;
+          std::tie(l, u) = to_interval(rat);
+          const double inf = std::numeric_limits<double>::infinity();
+
+          if (l == +inf) {
```

Comment:
Uh? It is a bug in to_interval (and should be fixed there) if it returns +inf as a lower bound.

Reply:
What I do currently is: I call `to_interval()` on `Type` that can be either `cpp_int` or `cpp_rational`. Inside `to_interval()`, I try to convert this `Type` number to double using `static_cast<double>(x)` and if it returns `+-inf`, I quit early and return `std::make_pair(+-inf, +-inf)`. This way I can check if I am in the limit case and return the proper interval. If I do not do that, then the call to `const int cmp = x.compare(i);` in the next lines of `to_interval()` fails because `i` is `+-inf`. I am not sure if this is a valid way to check it though.

Reply:
to_interval is an official interface, and (+inf,+inf) is not a valid return value for it. If casting to double gives +inf, the correct return value is (dbl_max, inf). You seem to do that anyway, just not in the right place.

Reply:
True. I changed this.